openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/UserRepository.java

```diff
@@ -583,6 +612,41 @@ private void updateRoles(User original, User updated) {
           ROLES_FIELD, origRoles, updatedRoles, added, deleted, EntityUtil.entityReferenceMatch);
     }
 
+    protected void updateDomain() {
+      if (operation.isPut() && !nullOrEmpty(original.getDomain()) && updatedByBot()) {
+        // Revert change to non-empty domain if it is being updated by a bot
+        // This is to prevent bots from overwriting the domain. Domain need to be
+        // updated with a PATCH request
+        updated.setUserDomains(original.getUserDomains());
+        return;
+      }
+
+      List<EntityReference> origDomains =
+          EntityUtil.populateEntityReferences(listOrEmptyMutable(original.getUserDomains()));
+      List<EntityReference> updatedDomains =
+          EntityUtil.populateEntityReferences(listOrEmptyMutable(updated.getUserDomains()));
+
+      // Remove Domains for the user
+      deleteTo(original.getId(), USER, Relationship.HAS, Entity.DOMAIN);
+
+      for (EntityReference domain : updatedDomains) {
+        addRelationship(domain.getId(), original.getId(), Entity.DOMAIN, USER, Relationship.HAS);
+      }
+
+      origDomains.sort(EntityUtil.compareEntityReference);
+      updatedDomains.sort(EntityUtil.compareEntityReference);
+
+      List<EntityReference> added = new ArrayList<>();
+      List<EntityReference> deleted = new ArrayList<>();
+      recordListChange(
+          "userDomains",
```

Comment:
we should just call it domain everywhere

Reply:
We cannot user and teams can have multiple domains , we can call it "domains" , since "domain" is already used feild and represents single entityRef , should I change it to domains?

Reply:
@mohityadav766 whats the difference between domain and domains here?
IMO i think we should restrict the team and user to be single domain or mark the single field to domains